homeassistant/components/cloud/http_api.py

```diff
@@ -225,6 +224,15 @@ async def post(self, request, data):
         hass = request.app["hass"]
         cloud = hass.data[DOMAIN]
         await cloud.login(data["email"], data["password"])
+
+        async def _on_start():
+            """Check if google assistant is linked."""
+            gconf = await cloud.client.get_google_config()
+            status = await gconf.async_sync_entities(gconf.agent_user_id)
+            if status == HTTP_OK:
+                await gconf.async_connect_agent_user(gconf.agent_user_id)
+
+        cloud.register_on_start(_on_start)
```

Comment:
I would expect this logic to be added in a more general way:

 - On Ha start, if connected: sync
 - On connected, if started: sync
 - When any sync entities request returns 200, connect the user agent ID

Reply:
We can indeed simplify it, if a sync completes it is already registered.

The only difference here is that we go around the `agent_user_ids` check, since that is still empty, it does not do a sync on connect.

Reply:
If you do want to do it here, you should make sure you de-register the `on_start`.